lib/salus/scanners/package_version/npm_package_version_scanner.rb

```diff
@@ -11,32 +11,58 @@ def should_run?
       @repository.package_lock_json_present?
     end
 
-    def check_for_violations(package_name, min_version, max_version)
+    def check_for_violations(package_name, min_version, max_version, blocked_versions)
       if @dependencies.key?(package_name)
         # repo_version: version used in the project
         repo_version = SemVersion.new(@dependencies[package_name].keys[0])
-        if repo_version && repo_version < min_version
-          line_number = @dependencies[package_name][repo_version.to_s].to_s
-          error_msg = "Package version for (#{package_name}) (#{repo_version})" \
-          "is less than minimum configured version (#{min_version}) on line " \
-          "{#{line_number}} in package-lock.json"
-          report_error(error_msg)
-          @passed = false
+
+        if repo_version
+          check_min_version(package_name, repo_version, min_version) if min_version.present?
+          check_max_version(package_name, repo_version, max_version) if max_version.present?
+          if blocked_versions.present?
+            check_blocked_versions(package_name, repo_version, blocked_versions)
+          end
         end
+      end
+    end
+
+    private
+
+    def check_min_version(package_name, repo_version, min_version)
+      if repo_version < min_version
+        line_number = @dependencies[package_name][repo_version.to_s].to_s
+        error_msg = "Package version for (#{package_name}) (#{repo_version})" \
+        "is less than minimum configured version (#{min_version}) on line " \
+        "{#{line_number}} in package-lock.json"
+        report_error(error_msg)
+        @passed = false
+      end
+    end
+
+    def check_max_version(package_name, repo_version, max_version)
```

Comment:
Maybe refactor `check_min_version` and `check_max_version`, because they share some code?

Reply:
If I remember correctly, you also need to `report_stderr`, because `report_error` doesn't display anything to the txt output, it only displays to the json.